pkg/apis/crds/karpenter.k8s.aws_ec2nodeclasses.yaml

```diff
@@ -57,24 +57,28 @@ spec:
                 EC2NodeClassSpec is the top level specification for the AWS Karpenter Provider.
                 This will contain configuration necessary to launch instances in AWS.
               properties:
-                amiFamily:
-                  description: AMIFamily is the AMI family that instances use.
-                  enum:
-                    - AL2
-                    - AL2023
-                    - Bottlerocket
-                    - Ubuntu
-                    - Custom
-                    - Windows2019
-                    - Windows2022
-                  type: string
                 amiSelectorTerms:
                   description: AMISelectorTerms is a list of or ami selector terms. The terms are ORed.
                   items:
                     description: |-
                       AMISelectorTerm defines selection logic for an ami used by Karpenter to launch nodes.
                       If multiple fields are used for selection, the requirements are ANDed.
                     properties:
+                      alias:
+                        description: |-
+                          Alias specifies which EKS optimized AMI to select.
+                          Each alias consistes of a family and a version, specified as "family@version".
+                          Valid families include: al2, al2023, bottlerocket, windows2019, and windows2022.
+                          The version can either be pinned to a specific AMI release, with that AMIs version format (ex: "al2023@v20240625" or "bottlerocket@1.10.0").
```

Comment:
Updated to include the v. I don't think it's necessary to document we support both, just documenting the v for consistency seems fine to me. I still want to support the version without the v though since that is the version string in the ssm path.